### DISCUSSION TITLE
Score reset, pack gen, and more!!

### DISCUSSION THREAD

**Discussion by MrPlayerYork** [RevonZev/minecraft-python-mcpy#8] [Ideas]
Was thinking of ways that would totally draw me into this project more and came up with a few ideas that you could consider adding.

- Support for resetting a score. I know you could just set it to 0, but I think this would be good for more experienced datapack creators.
- Pack generator. Not much to say other than maybe a way to have the program set up a datapack based on a pack.json file?
- A vscode code addon that would highlight syntax, give suggestions, and auto-complete would be amazing!!
- More versatile matches syntax. This one will take a bit of explaining:
  "Matches" is great but what would make it a million times better is if you could make commands based on the array of numbers. Ex:

**Keep this functionality**
```
unless score @a home matches [0, 2, 4]:
    say your score is 0 or 2 or 4

# Compile into:
execute unless score @a home matches 0 run say your score is 0 or 2 or 4
execute unless score @a home matches 2 run say your score is 0 or 2 or 4
execute unless score @a home matches 4 run say your score is 0 or 2 or 4
```
**But also allow this!!**
```
unless score @a home matches [0, 2, 4]:
    say your score is 0
    say your score is 2
    say your score is 4

# Compile into:
execute unless score @a home matches 0 run say your score is 0
execute unless score @a home matches 2 run say your score is 2
execute unless score @a home matches 4 run say your score is 4
```
To pull this string a little further, you also make it so we could reference the array. That we only have to change one part of the code and not 4 parts.
```
unless score @a home matches {array1=[0, 2, 4]}:
    say your score is ^array1^

# Compile into:
execute unless score @a home matches 0 run say your score is 0
execute unless score @a home matches 2 run say your score is 2
execute unless score @a home matches 4 run say your score is 4
```
(This is similar to defining functions)

I Will update/comment on this post if I can think of any other ideas!!

**Comment by revonzev** [comment]
- [x] Resetting score [v2.2.0](https://github.com/RevonZev/minecraft-python-mcpy/releases/tag/v2.2.0)  #9
```
reset player objective
```
- [Pack generator](https://www.planetminecraft.com/data-pack/datapack-generator/)
- I don't know how to make vscode addon yet. I use language-mcfunction vscode addon by Arcensoth by including `.mcpy` in `files.associations` in vscode's `settings.json`
```
"files.associations": {
        "*.mcpy": "mcfunction"
    }
```
- [x] For loop [v2.3.0](https://github.com/RevonZev/minecraft-python-mcpy/releases/tag/v2.3.0) #10 
```
for $i in [0, 1, 2]:
    say $i

for $x in [Hello, Hi, Heyo]:
    say $x

# Compile into:
say 0
say 1
say 2
say Hello
say Hi
say Heyo
```
This will also work with both of your suggestion
```
for $i in [0, 2, 4]:
    unless score @a home matches $i:
        say your score is 0 or 2 or 4

for $i in [1, 3, 5]:
    unless score @a home matches $i:
        say your score is $i
```
Then I'll deprecate `unless score @a home matches [0, 2, 4]:`
